File: IT/1/Oggetti/Esercizi/Java/Esercizio15.java
/* file Esercizio15.java */

class Engine {
	private int cc;

	public Engine(int cc) {
		this.cc = cc;
	}
}

class Car {
	private String model;
	private String vin;
	private String plate;
	private Engine engine;

	public Car(String model, String vin, String plate, Engine engine) {
		this.model = model;
		this.vin = vin;
		this.plate = plate;
		this.engine = engine;
	}

	public String getModel() {
		return this.model;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getPlate() {
		return this.plate;
	}

	public void setPlate(String plate) {
		this.plate = plate;
	}

	public String getVin() {
		return this.vin;
	}

	public void setVin(String vin) {
		this.vin = vin;
	}

	public Engine getEngine() {
		return this.engine;
	}

	public void setEngine(Engine engine) {
		this.engine = engine;
	}
}

File: IT/1/Oggetti/Esercizi/Python/esercizio15.py
# file esercizio15.py

class Engine(object):
	def __init__(self, cc):
		self.__cc = cc

class Car(object):
	def __init__(self, model, vin, plate, engine):
		self.model = model
		self.vin = vin
		self.plate = plate
		self.engine = engine

	def getModel(self):
		return self.__model

	def setModel(self, model):
		self.__model = model

	def getVin(self):
		return self.__vin

	def setVin(self, vin):
		self.__vin = vin

	def getPlate(self):
		return self.__plate

	def setPlate(self, plate):
		self.__plate = plate

	def getEngine(self):
		return self.__engine

	def setEngine(self, engine):
		self.__engine = engine

	model = property(getModel, setModel)
	vin = property(getVin, setVin)
	plate = property(getPlate, setPlate)
	engine = property(getEngine, setEngine)

